### ISSUE TITLE
Specify netCDF variable names in global config file

### ISSUE THREAD

**Issue by jhamman** [UW-Hydro/VIC#211]
In the current image mode driver, we have hard coded the variable names for forcing variables, domain variables, and parameter variables.  This includes netCDF dimension names.  As I have just moved to a new test case, I've had to modify these hard coded variable names, for example:
```diff
-    global_domain->n_nx = get_nc_dimension(nc_name, "ni");
-    global_domain->n_ny = get_nc_dimension(nc_name, "nj");
+    global_domain->n_nx = get_nc_dimension(nc_name, "lon");
+    global_domain->n_ny = get_nc_dimension(nc_name, "lat");
-    get_nc_field_int(nc_name, "run_cell", d2start, d2count, run);
+    get_nc_field_int(nc_name, "mask", d2start, d2count, run);
-    get_nc_field_double(nc_name, "xc",
+    get_nc_field_double(nc_name, "lon",
-    get_nc_field_double(nc_name, "yc",
+    get_nc_field_double(nc_name, "lat",
```

Would it make sense to add these field names to the global parameter file?  This is probably most important for the dimension names and for the forcing variables.  Parameter variables and domain file variables are mostly fixed and I don't think we need to provide too much flexibility in that way.  

@bartnijssen - The changes in the above diff were required to run the Stehekin example after the commits in #210.

**Comment by bartnijssen** [comment]
They should all be configurable (including dimension names). That is, they should be read from a config file as a dictionary.

**Comment by jhamman** [comment]
cc @wietsefranssen

**Comment by jhamman** [comment]
my proposal for how we should do this in the global parameter file is that we specify the netCDF variable name after listing the variable identifier e.g. `FORCE_TYPE    <var_identifier>    <nc_var_name>`.  In practice, this would look like this:

```
FORCING1      (put the forcing path/prefix here  # Forcing file path and prefix, ending in "_"
FORCE_TYPE    AIR_TEMP 	   tas    # Average air temperature, C
FORCE_TYPE    PREC 	       prcp   # Total precipitation (rain and snow), mm
FORCE_TYPE    PRESSURE 	   pres   # Atmospheric pressure, kPa
FORCE_TYPE    SHORTWAVE    dswrf  # Incoming shortwave, W/m2
FORCE_TYPE    LONGWAVE 	   dslwr  # Incoming longwave radiation, W/m2
FORCE_TYPE    QAIR 	       shum   # Vapor pressure, kPa
FORCE_TYPE    WIND 	       wind   # Wind speed, m/s
WIND_H        10.0                # height of wind speed measurement (m)
```

**Comment by tbohn** [comment]
Looks good to me.

**Comment by jhamman** [comment]
okay, I'll do this then.  I will be adding a member to the `force_type_struct` structure in `driver/shared`.

**Comment by tbohn** [comment]
Regarding the other netcdf input files (parameters, state vars), how/where should we define the names, and do you anticipate getting to this any time soon?

I'm simply asking because I'm looking at the names we're giving the parameter variables in the "tonic" scripts and thinking that they could stand to be a little clearer.  If we're going to make this totally flexible, then I can just go ahead and change the names that tonic gives to these variables without worrying about syncing this up with vic_init(), right?  But if this flexibility isn't going to happen soon, I'll keep the variable names as they are in vic_init().

**Comment by jhamman** [comment]
I think being flexible on the variable names would be okay but I'm not going to do that in this PR.  For now, I'd say make the names in the parameter file descriptive and change them in `vic_init` and `tonic`. Since he parameter file is very VIC specific, it is probably safe to assume folks can handle matching the required variable names. This is not the case for the forcing/domain files.

**Comment by tbohn** [comment]
OK, will do.  Shall I open a separate issue for these parameter name changes?

**Comment by jhamman** [comment]
a PR is fine.  Just reference this discussion there.

**Comment by wietsefranssen** [comment]
`FORCE_TYPE <var_identifier> <nc_var_name>`looks good for me too

**Comment by jhamman** [comment]
closed by #379
